### ISSUE TITLE
Package not specified error when creating a release

### ISSUE THREAD

**Issue by slewis74** [OctopusDeploy/Issues#5003]
When creating a release via `octo.exe` an error can occur where the package id isn't determined correctly for a step. The error surfaces as

> No package version was specified for _step_ : _package_

I was able to reproduce the issue, but only once. When I tried a second time it worked. As the attached screenshot show, when it worked the second time the release plan from the server had included the Package Name, which is presumably related to the root cause.

Customers have reported the issue when using the TFS/VSTS extension. I was able to repro on the command line so do not think the extension is the cause.

### Screen capture

![image](https://user-images.githubusercontent.com/4229735/47191660-f8a0c200-d38b-11e8-8ea9-0baf7f2add51.png)

## Affected versions

2018.8.x

**Octopus Server:** 

## Links

https://secure.helpscout.net/conversation/
https://secure.helpscout.net/conversation/688293304/33763?folderId=909887

**Comment by slewis74** [comment]
We now have an accurate repro for this issue. It occurs with earlier versions of Octo.exe calling the new server, due to an edge case we had not foreseen.

In my earlier testing, from the attached screenshot, I realised that the first run I did was using exe compiled against an earlier version of the code (even though I had the new code at the time). I'd run in debug then and so new binaries and it worked as expected every time after that. Rolled the code back and could repro every time.

So the fix is we need to update the build server extensions to have the new `octo.exe`
